data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/PipelineConnectorTest.java

```diff
@@ -78,6 +184,42 @@ public void testOutputSuccess() throws Exception {
         verify(buffer).write(eq(RECORD), anyInt());
     }
 
+    @Test
+    public void testEventBufferOutputSuccess() throws Exception {
+        eut.start(eventBuffer);
+
+        eut.output(eventRecordList);
+
+	Map.Entry<Collection<Record<Event>>, CheckpointState> ent = eventBuffer.read(1);
+	ArrayList<Record<Event>> records = new ArrayList<>(ent.getKey());
+	// Make sure the records are different
+	Assertions.assertNotEquals(eventRecordList.get(0),  records.get(0));
+	// Make sure the events are different
+	Event event1 = eventRecordList.get(0).getData();
+	Event event2 = records.get(0).getData();
+	Assertions.assertNotEquals(event1, event2);
+	event1.toMap().forEach((k, v)-> Assertions.assertEquals(event2.get(k, String.class), v));
+	event1.toMap().forEach((k, v)-> Assertions.assertEquals(k, testKey));
+	event1.toMap().forEach((k, v)-> Assertions.assertEquals(v, testValue));
+	
+    }
+
+    @Test
+    public void testSpanBufferOutputSuccess() throws Exception {
+        sput.start(spanBuffer);
+
+        sput.output(spanRecordList);
+
+	Map.Entry<Collection<Record<JacksonSpan>>, CheckpointState> ent = spanBuffer.doRead(10000);
+	ArrayList<Record<JacksonSpan>> records = new ArrayList<>(ent.getKey());
+	// Make sure the records are different
+	Assertions.assertNotEquals(spanRecordList.get(0),  records.get(0));
+	// Make sure the spans are different
+	JacksonSpan span1 = spanRecordList.get(0).getData();
+	JacksonSpan span2 = records.get(0).getData();
+	Assertions.assertNotEquals(span1, span2);
+    }
```

Comment:
This test should assert that the values are the same. Similar to your other test (lines 201-203).